src/AuditTrailModule/javasource/audittrail/log/CreateLogObject.java

```diff
@@ -296,115 +292,141 @@ private static IMendixObject createSingleLogLine(IMendixObject logObject, IMendi
+		if (ID == null) return Collections.emptyList();
+
+		// Get id of object
+		IMendixObject refObj = Core.retrieveId(currentcontext, ID);
+		List<IMendixObject> logLineList = new ArrayList<IMendixObject>();
+
+		if (refObj != null) {
+			// Get list of members with descriptive values
+			Collection<? extends IMendixObjectMember<?>> list = refObj.getMembers(currentcontext).values();
+
+			IMendixObject referenceLog = Core.instantiate(sudocontext, ReferenceLog.getType());
+			referenceLog.setValue(sudocontext, ReferenceLog.MemberNames.AttributeID.toString(), ID.toString());
+			referenceLog.setValue(sudocontext, ReferenceLog.MemberNames.Operation.toString(), typeOfReference.toString());
+			referenceLog.setValue(sudocontext, ReferenceLog.MemberNames.ReferenceLog_LogLine.toString(), parentID);
+			
+			logLineList.add(referenceLog);
+
+			if (list.size() > 0) {
+				// loop the list
+				for (IMendixObjectMember<?> member : list) {
+					IMendixObject referenceLogLine = Core.instantiate(sudocontext, ReferenceLogLine.getType());
+					referenceLogLine.setValue(sudocontext, ReferenceLogLine.MemberNames.Member.toString(), member.getName());
+					referenceLogLine.setValue(sudocontext, ReferenceLogLine.MemberNames.Value.toString(), getValue(member, true, currentcontext));
+					referenceLogLine.setValue(sudocontext, ReferenceLogLine.MemberNames.ReferenceLogLine_ReferenceLog.toString(), referenceLog.getId());
+					logLineList.add(referenceLogLine);
 				}
 			}
 		}
 
-		return value;
+		return logLineList;
 	}
+	
+	private static Comparator<IMendixIdentifier> IDCOMPARATOR = (IMendixIdentifier i1, IMendixIdentifier i2) -> (int)(i1.toLong() - i2.toLong());
 
+	// TODO The lists may be null. Check that.
 	@SuppressWarnings("unchecked")
-	private static IMendixObject createReferenceSetLogLine(IMendixObject logObject, IMendixObjectMember<?> member,
+	private static List<IMendixObject> createReferenceSetLogLine(IMendixObject logObject, IMendixObjectMember<?> member,
 			boolean isNew, IContext sudocontext, IContext currentcontext) throws CoreException {
-		String currentValue = "", previousValue = "";
 
 		List<IMendixIdentifier> currentIDList = (List<IMendixIdentifier>) member.getValue(currentcontext);
 		List<IMendixIdentifier> previousIDList = (List<IMendixIdentifier>) member.getOriginalValue(currentcontext);
+		
+		currentIDList.sort(IDCOMPARATOR);
+		previousIDList.sort(IDCOMPARATOR);
 
-		if (currentIDList != null && currentIDList.size() > 0) {
-			for (IMendixIdentifier id : currentIDList) {
-				currentValue += getValueFromReference(id, currentcontext);
-			}
-		}
-
-		if (previousIDList != null && previousIDList.size() > 0) {
-			for (IMendixIdentifier id : previousIDList) {
-				previousValue += getValueFromReference(id, currentcontext);
-			}
-		}
+		if (IncludeOnlyChangedAttributes == false || !currentIDList.equals(previousIDList) || isNew) {

```

Comment:
For null check safety :
```suggestion
              if (IncludeOnlyChangedAttributes == false || Objects.equal(currentIDList, previousIDList) || isNew) {
```

Reply:
Done